WordPress/gutenberg
# Components: Introduce a basic ProgressBar component

## What?
This PR introduces a basic horizontal `ProgressBar` component.

I'd love to get some design feedback since this is done from scratch and is using some component package colors by default, for consistency with the rest of the components. We're introducing some props to allow easier changing of the colors. 

## Why?
While it can be useful in various places, we're aiming to use this new component to replace the `Spinner` component that is currently utilized in the site editor loading experience. I've started work on this in #53032.

## How?
We're introducing a basic component with tests, stories, and README - all the basic pieces for a new component.

## Testing Instructions
* Run `npm run storybook:dev`
* Navigate to http://localhost:50240/?path=/docs/components-experimental-progressbar--default
* Play with the controls and verify the progress bar works well.
* Verify all tests pass.

### Testing Instructions for Keyboard
None

## Screenshots or screencast 

Indeterminate:

https://github.com/WordPress/gutenberg/assets/8436925/6d5fe03f-f420-4681-92e5-9fda885ce111

Determinate:

https://github.com/WordPress/gutenberg/assets/8436925/d868e20d-8430-4a20-8d48-85fb4854f38a

## ✍️ Dev Note

**Introducing a new experimental `ProgressBar` component**

A new experimental `ProgressBar` component has been introduced to the `@wordpress/components` package. 

The `ProgressBar` component supports two modes: 
* determinate - when a particular progress value has been specified (from 0 to 100)
* indeterminate - when a value hasn't been specified

The component is capable of switching between modes depending on the progress `value` prop that's passed to it.

The primary need and motivation for creating this new component is currently in the Site Editor, where it's already utilized for displaying the loading progress of the editor.
